packages/@lwc/engine-core/src/framework/mutation-tracker.ts

```diff
@@ -28,10 +29,29 @@ export function componentValueMutated(vm: VM, key: PropertyKey) {
     }
 }
 
-export function componentValueObserved(vm: VM, key: PropertyKey) {
+export function componentValueObserved(vm: VM, key: PropertyKey, target: any = {}) {
+    const { component, tro } = vm;
     // On the server side, we don't need mutation tracking. Skipping it improves performance.
     if (process.env.IS_BROWSER) {
-        valueObserved(vm.component, key);
+        valueObserved(component, key);
+    }
+
+    // The portion of reactivity that's exposed to signals is to subscribe a callback to re-render the VM (templates).
+    // We check check the following to ensure re-render is subscribed at the correct time.
+    //  1. The template is currently being rendered (there is a template reactive observer)
+    //  2. There was a call to a getter to access the signal (happens during vnode generation)
+    if (
+        lwcRuntimeFlags.ENABLE_EXPERIMENTAL_SIGNALS &&
+        target &&
+        typeof target === 'object' &&
```

Comment:
```suggestion
        isObject(target) &&
        !isNull(target) &&
```

Just our convention. Probably a perf voodoo dance, but let's just be consistent.